virtual_ship/sailship.py

```diff
@@ -222,68 +86,348 @@ def sailship(config: VirtualShipConfig):
     print("Simulating argo floats")
     simulate_argo_floats(
         out_path=os.path.join("results", "argo_floats.zarr"),
-        argo_floats=argo_floats,
+        argo_floats=schedule_results.measurements_to_simulate.argo_floats,
         fieldset=config.argo_float_config.fieldset,
         outputdt=timedelta(minutes=5),
         endtime=None,
     )
 
-    # convert CTD data to CSV
-    print("Postprocessing..")
-    postprocess()
-
-    print("All data has been gathered and postprocessed, returning home.")
-
+    # calculate cruise cost
+    assert (
+        config.waypoints[0].time is not None
+    ), "First waypoints cannot have None time as this has been verified before during config verification."
+    time_past = schedule_results.end_spacetime.time - config.waypoints[0].time
     cost = costs(config, time_past)
     print(f"This cruise took {time_past} and would have cost {cost:,.0f} euros.")
 
 
-def shiproute(config: VirtualShipConfig, dt: timedelta) -> list[Location]:
+def _simulate_schedule(
+    waypoints: list[Waypoint],
+    projection: pyproj.Geod,
+    config: VirtualShipConfig,
+) -> _ScheduleResults:
     """
-    Take in route coordinates and return lat and lon points within region of interest to sample.
+    Simulate the sailing and aggregate what measurements should be simulated.
 
+    :param waypoints: The schedule.
+    :param projection: Projection used to sail between waypoints.
```

Comment:
Also mention what the default is?

Reply:
I am not sure what you mean? this function does not provide a default?

Reply:
Well, I think the default is `pyproj.Geod(ellps="WGS84")`? Mention that here in the docstring?

Reply:
This function itself does not provide a default. The projection is passed when called the function. You might be referring to the sailship function, which indeed uses that projection.